File: src/main/Interpreter.java
package main;

import util.*;

public class Interpreter {

	public static void main(String[] args) {
		if(args.length == 0) {
			System.out.println("Please provide an input file to run.");
			return;
		}
			
		try {
			LexNode rootNode = new LexNode();
			LexAnalyzer analyzer = new LexAnalyzer(new TokenProvider(args[0]));
			
			analyzer.parse(rootNode, 0);
			
			
		} catch (Exception e) {
			System.out.println("Parsing broke! Something went wrong...");
			e.printStackTrace();
		}
	}

}

File: src/Interpreter.py
import sys
from util import LexNode, LexAnalyzer

def main():
	input = ""
	try:
		input = sys.argv[1]
	except(IndexError):
		print("Please provide an input file.")
		sys.exit()
	try:
		root_node = LexNode()
		analyzer = LexAnalyzer(TokenProvider(input))
		analyzer.parse(root_node, 0)
	except:
		Print("Parsing broke, something went wrong.")

if(__name__ == "__main__"):
	main()